contracts/WitnetRequestBoard.sol

```diff
@@ -148,23 +134,23 @@ contract WitnetRequestBoard is WitnetRequestBoardInterface {
 
     /// @dev Reports the result of a data request in Witnet.
     /// @param _id The unique identifier of the data request.
-    /// @param _drHash The unique hash of the request.
+    /// @param _drTxHash The unique hash of the request.
     /// @param _result The result itself as bytes.
     function reportResult(
         uint256 _id,
-        uint256 _drHash,
+        uint256 _drTxHash,
         bytes calldata _result
     ) external isAuthorized() validId(_id) resultNotIncluded(_id) {
-        require(_drHash != 0, "Data request transaction cannot be zero");
+        require(_drTxHash != 0, "Data request transaction cannot be zero");
         // Ensures the result byes do not have zero length
         // This would not be a valid encoding with CBOR and could trigger a reentrancy attack
         require(_result.length != 0, "Result has zero length");
 
-        requests[_id].drHash = _drHash;
+        requests[_id].drTxHash = _drTxHash;
         requests[_id].result = _result;
-        msg.sender.transfer(requests[_id].reward);
+        payable(msg.sender).transfer(requests[_id].reward);
```

Comment:
As long as a contract inheriting from this `WitnetBoardRequest` remembers to include `resultNotIncluded(_id)` (or similar verification), no reentrancy-attack is possible. However, it would be safer to set `requests[_id].reward = 0` before making the transfer. **Cons**: (a) one more word written to storage; (b) once paid, reward gets "forgotten".

Reply:
Events had better be emitted before transfer call.

Reply:
I think since at the moment only authorized addresses can call this function there is no reentracy attack to  consider, and besides, it can be interesting to check the rewards of the first requests of the bridge

Reply:
True, better not touch `reward` by now. Have in mind, however, there is no actual method at the moment as to externally read `DataRequest.requestAddress`, `DataRequest.drOutputHash` or 'DataRequest.reward'.